### PR TITLE
Add listener for a player firing an arrow

### PR DESCRIPTION
**Pull request by Restioson**
Crossbow support is not there quite yet, but I've tested bow support in koth and cancellation works.

### CODE REVIEW

**Comment by Restioson** [comment]
It's more difficult to write a mixin for crossbow since it uses static methods to dispatch this, and they're quite broken up, so I'm not sure where to inject the code. For example, it calls `shootAll` which has the shooting logic and then `setCharged` afterward. Ideally returning `ActionResult.FAIL` would not reset the charge on the crossbow, so it would need to be mixed in to before the `shootAll` call. However, by this time none of the arrow entities have been created, so no useful information can be passed forward.

File: src/main/java/xyz/nucleoid/plasmid/game/event/PlayerFireArrowListener.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * <li>{@link ActionResult#FAIL} cancels further handlers and does not execute vanilla behavior.
+ * <li>{@link ActionResult#PASS} moves on to the next listener.</ul>
+ */
+public interface PlayerFireArrowListener {
+    EventType<PlayerFireArrowListener> EVENT = EventType.create(PlayerFireArrowListener.class, listeners -> (user, tool, arrows, remaining, projectile) -> {
+        for (PlayerFireArrowListener listener : listeners) {
+            ActionResult result = listener.onFireArrow(user, tool, arrows, remaining, projectile);
+
+            if (result != ActionResult.PASS) {
+                System.out.println("Returning fail");
```
Comment (Gegy) [inline src/main/java/xyz/nucleoid/plasmid/game/event/PlayerFireArrowListener.java:26]: Debug print :P

Comment (Restioson) [reply src/main/java/xyz/nucleoid/plasmid/game/event/PlayerFireArrowListener.java:26]: Oof I thought this printed from test minigame 😳

**Comment by Gegy** [comment]
I think it's fair to merge this PR without support for crossbows, since the event signature shouldn't have to change later if we were to add support for them.

**Comment by Restioson** [comment]
> I think it's fair to merge this PR without support for crossbows, since the event signature shouldn't have to change later if we were to add support for them.

Alright, I'll mark it ready for review and then we can open an issue for crossbows.

**Pull request merged**